pkg/sql/advisorylock/advisorylock_test: TestAdvisoryLockSQLDeadlock failed

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://mesolite.cluster.engflow.com/invocations/default/0f61d737-e629-48b1-b0f7-b6ad54307593?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2Fkdmlzb3J5bG9jazphZHZpc29yeWxvY2tfdGVzdA==) on master @ [399a564350740bb9d588321c910ecd6c03a87153](https://github.com/cockroachdb/cockroach/commits/399a564350740bb9d588321c910ecd6c03a87153):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestAdvisoryLockSQLDeadlock1741254933
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:639: DRPC is enabled (via testserver factory defaults)
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestAdvisoryLockSQLDeadlock1741254933
--- FAIL: TestAdvisoryLockSQLDeadlock (0.76s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63132

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21323057?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21323057?buildTab=artifacts#/) on master @ [17bc3d7fba8d5048ff334b95f041f8369e2025b4](https://github.com/cockroachdb/cockroach/commits/17bc3d7fba8d5048ff334b95f041f8369e2025b4):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock3136880149
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock3136880149
--- FAIL: TestAdvisoryLockSQLDeadlock (0.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21323161?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21323161?buildTab=artifacts#/) on master @ [0b20333edc0ce6891f02f630d2729b7d96d2ef67](https://github.com/cockroachdb/cockroach/commits/0b20333edc0ce6891f02f630d2729b7d96d2ef67):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock1391307237
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock1391307237
--- FAIL: TestAdvisoryLockSQLDeadlock (1.66s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://mesolite.cluster.engflow.com/invocations/default/a5a83445-64a4-465c-9220-d57150d4cbb8?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2Fkdmlzb3J5bG9jazphZHZpc29yeWxvY2tfdGVzdA==) on master @ [f8ca34e31c8f9d2d72847018d9296970eaea5364](https://github.com/cockroachdb/cockroach/commits/f8ca34e31c8f9d2d72847018d9296970eaea5364):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestAdvisoryLockSQLDeadlock3079717688
    test_log_scope.go:82: use -show-logs to present logs inline
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestAdvisoryLockSQLDeadlock3079717688
--- FAIL: TestAdvisoryLockSQLDeadlock (2.28s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://mesolite.cluster.engflow.com/invocations/default/e2b4d4ad-ae50-4615-9079-6432fc08e5c4?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2Fkdmlzb3J5bG9jazphZHZpc29yeWxvY2tfdGVzdA==) on master @ [f8ca34e31c8f9d2d72847018d9296970eaea5364](https://github.com/cockroachdb/cockroach/commits/f8ca34e31c8f9d2d72847018d9296970eaea5364):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestAdvisoryLockSQLDeadlock540267285
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestAdvisoryLockSQLDeadlock540267285
--- FAIL: TestAdvisoryLockSQLDeadlock (7.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21325184?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21325184?buildTab=artifacts#/) on master @ [37d4060c3288f7ef326bb43e1785d9790c0f73bd](https://github.com/cockroachdb/cockroach/commits/37d4060c3288f7ef326bb43e1785d9790c0f73bd):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock3982817446
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock3982817446
--- FAIL: TestAdvisoryLockSQLDeadlock (2.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://mesolite.cluster.engflow.com/invocations/default/bc54c7ab-795c-412c-bd4e-e9e1af92a6f6?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2Fkdmlzb3J5bG9jazphZHZpc29yeWxvY2tfdGVzdA==) on master @ [eb8f019262c438d83266cf3e1d668a00ff6399d8](https://github.com/cockroachdb/cockroach/commits/eb8f019262c438d83266cf3e1d668a00ff6399d8):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestAdvisoryLockSQLDeadlock1503577738
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestAdvisoryLockSQLDeadlock1503577738
--- FAIL: TestAdvisoryLockSQLDeadlock (7.28s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://mesolite.cluster.engflow.com/invocations/default/db20bf7e-d083-4be9-85ed-b900a57310c1?testReportRun=6&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2Fkdmlzb3J5bG9jazphZHZpc29yeWxvY2tfdGVzdA==) on master @ [eb8f019262c438d83266cf3e1d668a00ff6399d8](https://github.com/cockroachdb/cockroach/commits/eb8f019262c438d83266cf3e1d668a00ff6399d8):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestAdvisoryLockSQLDeadlock1917291306
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestAdvisoryLockSQLDeadlock1917291306
--- FAIL: TestAdvisoryLockSQLDeadlock (1.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://mesolite.cluster.engflow.com/invocations/default/3feab18f-c5a6-418f-bf98-963710966019?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2Fkdmlzb3J5bG9jazphZHZpc29yeWxvY2tfdGVzdA==) on master @ [5810d94e7858abf37a92463895123958e30e60ee](https://github.com/cockroachdb/cockroach/commits/5810d94e7858abf37a92463895123958e30e60ee):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestAdvisoryLockSQLDeadlock427039850
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestAdvisoryLockSQLDeadlock427039850
--- FAIL: TestAdvisoryLockSQLDeadlock (3.57s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21328220?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21328220?buildTab=artifacts#/) on master @ [65d2e28a2adf6ec3c5bc57d8e668abb943a9a4db](https://github.com/cockroachdb/cockroach/commits/65d2e28a2adf6ec3c5bc57d8e668abb943a9a4db):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock51501883
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock51501883
--- FAIL: TestAdvisoryLockSQLDeadlock (1.87s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21328701?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21328701?buildTab=artifacts#/) on master @ [ff6df8c82e29ebb7217f0222d84822e2e3547009](https://github.com/cockroachdb/cockroach/commits/ff6df8c82e29ebb7217f0222d84822e2e3547009):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock501329015
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock501329015
--- FAIL: TestAdvisoryLockSQLDeadlock (1.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21329008?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21329008?buildTab=artifacts#/) on master @ [544fcd7b947c2e0160d162a225984d4dcbf3da5d](https://github.com/cockroachdb/cockroach/commits/544fcd7b947c2e0160d162a225984d4dcbf3da5d):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock1743785837
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock1743785837
--- FAIL: TestAdvisoryLockSQLDeadlock (1.51s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21329328?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21329328?buildTab=artifacts#/) on master @ [3399ae7c9d6e7b7c840a2355f9fbc50a4cf28ffe](https://github.com/cockroachdb/cockroach/commits/3399ae7c9d6e7b7c840a2355f9fbc50a4cf28ffe):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock3096914584
    test_log_scope.go:82: use -show-logs to present logs inline
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock3096914584
--- FAIL: TestAdvisoryLockSQLDeadlock (0.76s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://mesolite.cluster.engflow.com/invocations/default/3029206d-8ad6-47e7-b13f-3c1bfda76083?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2Fkdmlzb3J5bG9jazphZHZpc29yeWxvY2tfdGVzdA==) on master @ [27f490bc59364b059b1dd4b274072e02251b1b20](https://github.com/cockroachdb/cockroach/commits/27f490bc59364b059b1dd4b274072e02251b1b20):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestAdvisoryLockSQLDeadlock1797498876
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestAdvisoryLockSQLDeadlock1797498876
--- FAIL: TestAdvisoryLockSQLDeadlock (7.95s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21330105?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21330105?buildTab=artifacts#/) on master @ [cef4bd40582c998b4fd1efaa8f0196755c3e0f26](https://github.com/cockroachdb/cockroach/commits/cef4bd40582c998b4fd1efaa8f0196755c3e0f26):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock1749767170
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock1749767170
--- FAIL: TestAdvisoryLockSQLDeadlock (1.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/advisorylock/advisorylock_test.TestAdvisoryLockSQLDeadlock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21330354?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21330354?buildTab=artifacts#/) on master @ [3fe3c2cc428b2674511a41ff3ebd2f31b8a10b80](https://github.com/cockroachdb/cockroach/commits/3fe3c2cc428b2674511a41ff3ebd2f31b8a10b80):


```
=== RUN   TestAdvisoryLockSQLDeadlock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock1740788743
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    advisory_lock_test.go:272: 
        	Error Trace:	pkg/sql/advisorylock/advisory_lock_test.go:272
        	Error:      	Should be true
        	Test:       	TestAdvisoryLockSQLDeadlock
        	Messages:   	expected at least one failing statement when breaking a lock cycle, got []error{error(nil), error(nil)}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fa95b1fcffd1e59b847c8efdf8577a71/logTestAdvisoryLockSQLDeadlock1740788743
--- FAIL: TestAdvisoryLockSQLDeadlock (1.30s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdvisoryLockSQLDeadlock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/advisorylock/advisory_lock_test.go
+++ b/pkg/sql/advisorylock/advisory_lock_test.go
@@ -32,6 +32,22 @@ func assertLockCycleBrokenError(t *testing.T, err error) {
 		"expected lock-cycle resolution to surface as a transaction retry error, got: %v", err)
 }
 
+// finishAdvisoryDeadlockTxn captures the error surfaced when the deadlock
+// cycle is broken. The losing side's failure can show up either on the
+// second pg_advisory_xact_lock call or at COMMIT — KV deadlock detection
+// force-aborts the victim, but the abort may only become visible to the
+// client at commit time if the lock acquisition slips past the abort (e.g.
+// because the lock holder released the lock before the abort propagated).
+// This helper tries COMMIT when the lock acquire returned nil so either
+// failure mode is captured on errCh.
+func finishAdvisoryDeadlockTxn(ctx context.Context, conn *gosql.Conn, lockErr error) error {
+	if lockErr != nil {
+		return lockErr
+	}
+	_, err := conn.ExecContext(ctx, `COMMIT`)
+	return err
+}
+
 // TestAdvisoryLockSQLAcquireRelease verifies locks are released on commit so a
 // later transaction can acquire the same key.
 func TestAdvisoryLockSQLAcquireRelease(t *testing.T) {
@@ -220,7 +236,7 @@ func TestAdvisoryLockSQLDeadlock(t *testing.T) {
 		select {
 		case start2 <- struct{}{}:
 		case <-ctx.Done():
-			return err
+			return ctx.Err()
 		}
 		select {
 		case <-proceed1:
@@ -230,7 +246,7 @@ func TestAdvisoryLockSQLDeadlock(t *testing.T) {
 		_, err := conn1.ExecContext(ctx, `SELECT pg_advisory_xact_lock($1)`, k2)
 		// Note: The final error cannot be returned here, since it will cancel the context
 		// if something failed.
-		errCh <- err
+		errCh <- finishAdvisoryDeadlockTxn(ctx, conn1, err)
 		return nil
 	})
 
@@ -254,7 +270,7 @@ func TestAdvisoryLockSQLDeadlock(t *testing.T) {
 			return ctx.Err()
 		}
 		_, err := conn2.ExecContext(ctx, `SELECT pg_advisory_xact_lock($1)`, k1)
-		errCh <- err
+		errCh <- finishAdvisoryDeadlockTxn(ctx, conn2, err)
 		return nil
 	})
 
@@ -270,7 +286,7 @@ func TestAdvisoryLockSQLDeadlock(t *testing.T) {
 		assertLockCycleBrokenError(t, err)
 	}
 	require.True(t, sawBreakingErr,
-		"expected at least one failing statement when breaking a lock cycle, got %#v", errs)
+		"expected at least one failing statement or commit when breaking a lock cycle, got %#v", errs)
 }
 
 // TestAdvisoryLockSQLDeadlockForUpdateCrossAdvisory builds a wait cycle between a
@@ -337,12 +353,6 @@ func TestAdvisoryLockSQLDeadlockForUpdateCrossAdvisory(t *testing.T) {
 			return err
 		}
 		defer func() { _, _ = conn1.ExecContext(context.Background(), `ROLLBACK`) }()
-		// Return at least one row so results are flushed to the client before the
-		// statements that participate in the deadlock. Otherwise explicit-txn
-		// auto-retry can rewind transparently and both Exec calls may return nil.
-		if _, err := conn1.ExecContext(ctx, `SELECT 1`); err != nil {
-			return err
-		}
 		if _, err := conn1.ExecContext(ctx,
 			`SELECT * FROM dead_adv_row WHERE id = 1 FOR UPDATE`); err != nil {
 			return err
@@ -356,7 +366,7 @@ func TestAdvisoryLockSQLDeadlockForUpdateCrossAdvisory(t *testing.T) {
 		_, err := conn1.ExecContext(ctx, `SELECT pg_advisory_xact_lock($1)`, advKey)
 		// Note: The final error cannot be returned here, since it will cancel the context
 		// if something failed.
-		errCh <- err
+		errCh <- finishAdvisoryDeadlockTxn(ctx, conn1, err)
 		return nil
 	})
 
@@ -370,15 +380,12 @@ func TestAdvisoryLockSQLDeadlockForUpdateCrossAdvisory(t *testing.T) {
 			return err
 		}
 		defer func() { _, _ = conn2.ExecContext(context.Background(), `ROLLBACK`) }()
-		if _, err := conn2.ExecContext(ctx, `SELECT 1`); err != nil {
-			return err
-		}
 		if _, err := conn2.ExecContext(ctx, `SELECT pg_advisory_xact_lock($1)`, advKey); err != nil {
 			return err
 		}
 		close(g2HasAdv)
 		_, err := conn2.ExecContext(ctx, `SELECT * FROM dead_adv_row WHERE id = 1 FOR UPDATE`)
-		errCh <- err
+		errCh <- finishAdvisoryDeadlockTxn(ctx, conn2, err)
 		return nil
 	})
 
@@ -394,5 +401,5 @@ func TestAdvisoryLockSQLDeadlockForUpdateCrossAdvisory(t *testing.T) {
 		assertLockCycleBrokenError(t, err)
 	}
 	require.True(t, sawBreakingErr,
-		"expected at least one failing statement when breaking FOR UPDATE vs advisory cycle, got %#v", errs)
+		"expected at least one failing statement or commit when breaking FOR UPDATE vs advisory cycle, got %#v", errs)
 }

--- a/pkg/sql/conn_executor.go
+++ b/pkg/sql/conn_executor.go
@@ -4092,6 +4092,7 @@ func (ex *connExecutor) initPlanner(ctx context.Context, p *planner) {
 	p.statsCollector = ex.statsCollector
 	p.sessionDataMutatorIterator = ex.dataMutatorIterator
 	p.noticeSender = nil
+	p.stmtResultBuffering = nil
 	p.preparedStatements = ex.getPrepStmtsAccessor()
 	p.sqlCursors = ex.getCursorAccessor()
 	p.routineMetadataForwarder = nil

--- a/pkg/sql/conn_executor_exec.go
+++ b/pkg/sql/conn_executor_exec.go
@@ -518,6 +518,7 @@ func (ex *connExecutor) execStmtInOpenState(
 	ex.resetPlanner(ctx, p, ex.state.mu.txn, stmtTS)
 	p.sessionDataMutatorIterator.ParamStatusUpdater = res
 	p.noticeSender = res
+	p.stmtResultBuffering = res
 	ih := &p.instrumentation
 
 	if ex.executorType != executorTypeInternal {
@@ -1455,6 +1456,7 @@ func (ex *connExecutor) execStmtInOpenStateWithPausablePortal(
 	ex.resetPlanner(ctx, p, ex.state.mu.txn, stmtTS)
 	p.sessionDataMutatorIterator.ParamStatusUpdater = res
 	p.noticeSender = res
+	p.stmtResultBuffering = res
 	ih := &p.instrumentation
 
 	if ex.executorType != executorTypeInternal {

--- a/pkg/sql/conn_io.go
+++ b/pkg/sql/conn_io.go
@@ -933,6 +933,12 @@ type RestrictedCommandResult interface {
 	RevokePortalPausability() error
 }
 
+// resultBufferingDisabler is the subset of RestrictedCommandResult used to
+// flush a statement's result writer mid-execution (see DisableBuffering).
+type resultBufferingDisabler interface {
+	DisableBuffering()
+}
+
 // DescribeResult represents the result of a Describe command (for either
 // describing a prepared statement or a portal).
 type DescribeResult interface {

--- a/pkg/sql/planner.go
+++ b/pkg/sql/planner.go
@@ -298,6 +298,13 @@ type planner struct {
 	// instead.
 	noticeSender noticeSender
 
+	// stmtResultBuffering exposes the current statement's result writer so a
+	// side-effecting builtin can flush its results immediately and prevent
+	// transparent connExecutor rewind across the side effect. Set in
+	// execStmtInOpenState after resetPlanner; nil between statements and for
+	// the internal executor.
+	stmtResultBuffering resultBufferingDisabler
+
 	queryCacheSession querycache.Session
 
 	// evalCatalogBuiltins is used as part of the eval.Context.
@@ -1350,6 +1357,13 @@ func (p *planner) advisoryXactLockImpl(
 			return false, nil
 		}
 		return false, err
+	}
+	// The acquire is a visible side effect; flush the statement's result so
+	// the connExecutor cannot transparently rewind past it on a subsequent
+	// serializable push (e.g. a deadlock break would otherwise be masked by
+	// auto-retry and the losing client would see no error).
+	if w := p.stmtResultBuffering; w != nil {
+		w.DisableBuffering()
 	}
 	return true, nil
 }

